crates/manifest/src/schema/common.rs

```diff
@@ -92,6 +92,27 @@ pub struct ComponentBuildConfig {
     pub watch: Vec<String>,
 }
 
+impl ComponentBuildConfig {
+    /// The commands to execute for the build
+    pub fn commands(&self) -> impl Iterator<Item = &String> {
+        let as_vec = match &self.command {
+            Commands::Single(cmd) => vec![cmd],
+            Commands::Multiple(cmds) => cmds.iter().collect(), // I know, but I haven't been able to find a less worse way to line up the types between the two arms
+        };
+        as_vec.into_iter()
```

Comment:
I don't think the comment adds anything here. I think we should delete it. Also, you can just clone instead of iter and collecting. 

```suggestion
        let as_vec = match &self.command {
            Commands::Single(cmd) => vec![cmd],
            Commands::Multiple(cmds) => cmds.clone(),
        };
        as_vec.into_iter()
```

Reply:
Cloning doesn't typecheck (`Vec<String>` vs `Vec<&String>`).  I can clone `cmd` in the single case too, if you like?

Reply:
@rylev Actually, since you kindly approved despite this, I'm going to merge with the code as is (I've deleted the comment as you suggested) and we can simplify it in a later PR - hope that's okay.